Apps/WebClient/src/Server/Models/WalletCredentialModel.cs

```diff
@@ -0,0 +1,97 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-------------------------------------------------------------------------
+namespace HealthGateway.WebClient.Models
+{
+    using System;
+    using HealthGateway.Database.Constants;
+    using HealthGateway.Database.Models;
+
+    /// <summary>
+    /// Model that provides a user representation of an WalletCredentialModel.
+    /// </summary>
+    public class WalletCredentialModel
+    {
+        /// <summary>
+        /// Gets or sets the verifiable credential id.
+        /// </summary>
+        public Guid CredentialId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the wallet connection id.
+        /// </summary>
+        public Guid WalletConnectionId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the source type.
+        /// </summary>
+        public string SourceType { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the source id.
+        /// </summary>
+        public string SourceId { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the issuer connection id.
+        /// </summary>
+        public Guid IssuerConnectionId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the connection date.
+        /// </summary>
+        public DateTime? AddedDate { get; set; }
+
+        /// <summary>
+        /// Gets or sets the disconnection date.
+        /// </summary>
+        public DateTime? RevokedDate { get; set; }
+
+        /// <summary>
+        /// Gets or sets the connection state.
+        /// </summary>
+        public WalletCredentialStatus State { get; set; }
+
+        /// <summary>
+        /// Gets or sets the db record version.
+        /// </summary>
+        public uint Version { get; set; }
+
+        /// <summary>
+        /// Constructs a WalletCredentialModel from a database model.
+        /// </summary>
+        /// <param name="walletCredential">The wallet credential database model.</param>
+        /// <returns>The verifiable credential model.</returns>
+        public static WalletCredentialModel CreateFromDbModel(WalletCredential walletCredential)
+        {
+            return new WalletCredentialModel()
+            {
+                CredentialId = walletCredential.Id,
+                WalletConnectionId = walletCredential.WalletConnectionId,
+
+                // TODO: IssuerConnectionId is duplicated between connection/credential, should we remove it?
+                IssuerConnectionId = walletCredential.WalletConnection.AgentId,
+
+                State = walletCredential.Status,
+                SourceId = walletCredential.ResourceId,
+                AddedDate = walletCredential.AddedDateTime,
+                RevokedDate = walletCredential.RevokedDateTime,
+                Version = walletCredential.Version,
+
+                // TODO: SourceType must be added to db model
```

Comment:
Should we add the source type to the db model?

Reply:
+1